skimage/segmentation/_chan_vese.py

```diff
@@ -0,0 +1,298 @@
+import numpy as np
+from scipy.ndimage import distance_transform_edt as distance
+
+
+def _cv_curvature(phi):
+    """
+    Returns the 'curvature' of a level set 'phi'.
+    """
+    P = np.pad(phi, 1, mode='edge')
+    fy = (P[2:, 1:-1] - P[0:-2, 1:-1]) / 2.0
+    fx = (P[1:-1, 2:] - P[1:-1, 0:-2]) / 2.0
+    fyy = P[2:, 1:-1] + P[0:-2, 1:-1] - 2*phi
+    fxx = P[1:-1, 2:] + P[1:-1, 0:-2] - 2*phi
+    fxy = .25 * (P[2:, 2:]+P[:-2, :-2]-P[:-2, 2:]-P[2:, :-2])
+    K = ((fxx*fy**2 - 2*fxy*fx*fy + fyy*fx**2) /
+         (np.power(fx**2 + fy**2, 1.5) + 1e-10))
```

Comment:
``x * np.sqrt(x**2)`` is quite faster than ``np.power(x, 1.5)``, as I found out when profiling your code.